docs/ops/state/large_state_tuning.md

```diff
@@ -210,6 +211,71 @@ and not from the JVM. Any memory you assign to RocksDB will have to be accounted
 of the TaskManagers by the same amount. Not doing that may result in YARN/Mesos/etc terminating the JVM processes for
 allocating more memory than configured.
 
+### Bounding RocksDB Memory Usage
+
+RocksDB allocates native memory outside of the JVM, which could lead the process to exceed the total memory budget.
+This can be especially problematic in containerized environments such as Kubernetes that kill processes who exceed their memory budgets.
+Flink limit total memory usage of RocksDB instance(s) per slot by leveraging shareable [cache](https://github.com/facebook/rocksdb/wiki/Block-Cache)
+and [write buffer manager](https://github.com/facebook/rocksdb/wiki/Write-Buffer-Manager) among all instances in a single slot by default.
+The shared cache will place an upper limit on the [three components](https://github.com/facebook/rocksdb/wiki/Memory-usage-in-RocksDB) that use the majority of memory
+when RocksDB is deployed as a state backend: block cache, index and bloom filters, and MemTables.
+This feature is enabled by default and could be controlled by two ways:
+  -  Integrate with managed memory of task manager: turn `state.backend.rocksdb.memory.managed` as true. If so, RocksDB state backend will use the managed memory budget of the task slot to set the capacity of that shared cache object.
+  This operation is enabled by default, which means Flink would always choose to integrate RocksDB memory usage with the managed memory first.
+  -  Not integrated with managed memory: configure the memory size of `state.backend.rocksdb.memory.fixed-per-slot` to set the fixed total amount of memory per slot.
+  This option will override `state.backend.rocksdb.memory.managed` option when configured and ignore calculated managed memory per slot from task manager.
+  User could also configure `taskmanager.memory.task.off-heap.size` to set additional quota in off-heap memory, which should be equal to `taskmanager.numberOfTaskSlots` * ``state.backend.rocksdb.memory.fixed-per-slot``, to fit in Flink's memory model.
```

Comment:
I think user could manually control memory usage via `ClumnFamilyOptions` even our feature of shared cache is not introduced in. What's more, after our feature is introduced, if `state.backend.rocksdb.memory.managed` is still `true` and the `ClumnFamilyOptions` which added by user would not take real effect since we will finally create a write buffer manager and a cache to share.
I think we might not need to talk about the `ClumnFamilyOptions`
I

Reply:
I think the context here is already not using the managed memory or `state.backend.rocksdb.memory.managed` is `false`. On the other hand, to make it explicit, we could also add a simple line like "*Experts Only*: To control memory manually instead of using managed memory, you can set `state.backend.rocksdb.memory.managed` to `false` and control through `ClumnFamilyOptions`. Or to save some manual calculation, .... "